theano/tensor/elemwise.py

```diff
@@ -1179,7 +1186,7 @@ def c_support_code_apply(self, node, nodename):
         return support_code
 
     def c_code_cache_version_apply(self, node):
-        version = [11]  # the version corresponding to the c code in this Op
+        version = [12]  # the version corresponding to the c code in this Op
```

Comment:
I don't think a single night of recompilation is "very bad".  However if you feel that the benefits really do not outweight the cost, I can change it back to 11.  The alternative of switching version depending on the dtype is worse than both to me.